### DISCUSSION TITLE
Multiple storage class on a same deployment

### DISCUSSION THREAD

**Discussion by KKWait** [minio/minio#18556] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Not a problem, just a question regarding a need we have.

**Describe the solution you'd like**
We're currently testing Minio to provide our customers a shared s3 infrastructure, so far backed with HDD disks. Some of them would like to be able to chose their performance tier when creating a bucket. Idealy, we would have a distributed setup (X hosts, Y HDD disks per hosts and Z SSD disks per hosts, 1 erasure set per disk topology). Then customers would be able to choose on which tier their data should be stored, and cherry on the cake, would be able to move data during its lifecycle, depending its needs.

**Describe alternatives you've considered**
Deploying 2 environments, 1rst with HDD dedicated disks and 2nd with SSD disks. But that's not optimised ....

**Additional context**
Nothing more

**Comment by klauspost** [comment, marked as answer]
MinIO uses deterministic placement, so objects cannot be arbitrarily placed. Also a MinIO cluster should provide uniform hardware. If you want different levels of performance, set them up independently.

You can configure different storage classes that provide different levels of redundancy.

You can configure [tiering](https://min.io/docs/minio/kubernetes/upstream/administration/object-management/object-lifecycle-management.html#object-transition-tiering) that offloads the data - but not metadata of objects after a defined time.

**Reply by KKWait** [reply]
Understood, thanks for your answer !
